### PR TITLE
Fix #9 : wstest 1.2.1-1.2.8 binary message support

### LINKED ISSUE #9: Solve wstest 1.2.1

### PR DESCRIPTION
**Pull request by Nek-**
image

This fixes #9 #10 #11 #12 #13 #14 #15 #16

### CODE REVIEW

File: src/Message/MessageHandlerInterface.php [hunk 1]
```diff
@@ -28,12 +28,20 @@
...
     /**
-     * Is called on new data.
+     * Is called on new text data.
      *
-     * @param string $data
+     * @param string     $data       Text data
      * @param Connection $connection
      */
-    public function onData($data, Connection $connection);
+    public function onMessage($data, Connection $connection);
```
Comment (folliked) [inline src/Message/MessageHandlerInterface.php]: no type ?

File: src/Message/MessageHandlerInterface.php [hunk 2]
```diff
@@ -28,12 +28,20 @@
...
+
+    /**
+     * Is called on new binary data.
+     *
+     * @param string     $data       Binary data
+     * @param Connection $connection
+     */
+    public function onBinary($data, Connection $connection);
```
Comment (folliked) [inline src/Message/MessageHandlerInterface.php]: no type ?

File: src/Rfc6455/Message.php [hunk 3]
```diff
@@ -111,6 +111,14 @@ public function getContent() : string
     }
 
     /**
+     * @return int
+     */
+    public function getOpcode()
+    {
+        return $this->getFirstFrame()->getOpcode();
```
Comment (folliked) [inline src/Rfc6455/Message.php]: no test getFirstFrame() is not null ?

Comment (Nek-) [inline src/Rfc6455/Message.php]: It's ok inside `getFirstFrame`

**Pull request merged**